.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseSetParser.cs

```diff
@@ -32,61 +33,78 @@ public DateTimeParseResult Parse(ExtractResult er, DateObject refDate)
 
             if (er.Type.Equals(ParserName, StringComparison.Ordinal))
             {
-                var innerResult = ParseEachUnit(er.Text);
+                var innerResult = ParseEachUnit(er.Text.Trim());
```

Comment:
Why are these trims needed? If really necessary, trim should be called only once.